spec/requests/demo_mode/sessions_spec.rb

```diff
@@ -72,6 +75,36 @@
           expect(response_json['password']).to be_nil
         end
       end
+
+      context 'with option' do
+        before do
+          DemoMode.configure do
+            around_persona_generation do |_, options|
+              if options.present? && options[:example_custom_option].present?
+                DummyUser.create!(name: options[:example_custom_option][:name])
+              end
+            end
+          end
+        end
+
+        let(:options) { { example_custom_option: { name: 'Tester' } } }
+
+        it 'creates a session and returns processing json saving the option on the created session' do
+          post '/ohno/sessions', params: {
+            session: { persona_name: 'the_everyperson' },
+            options: options,
+          }.to_json, headers: request_headers
+
+          last_session = DemoMode::Session.last
+          DemoMode::AccountGenerationJob.perform_now(last_session, **options)
+
+          expect(DummyUser.last.name).to eq 'Tester'
+          expect(response_json['id']).to eq last_session.id
+          expect(response_json['processing']).to be true
+          expect(response_json['username']).to be_nil
+          expect(response_json['password']).to be_nil
+        end
+      end
```

Comment:
@smudge @pat-whitrock let me know if this makes sense. I have to do the `DemoMode::AccountGenerationJob.perform_now` on line 99 to get the persona generated in time

Reply:
I left a suggestion above about triggering the enqueued job to be executed via the queue adapter API, rather than creating an entirely new job to be synchronously performed.

> I have to do the `DemoMode::AccountGenerationJob.perform_now` on line 99 to get the persona generated in time

I don't think you were running into a race condition. In the test environment, the enqueued job will sit in the queue until you explicitly instruct the queue to process it. But you're still right that we need to do something between the HTTP request and the assertions.